hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/UpsertPartitioner.java

```diff
@@ -170,7 +171,7 @@ private void assignInserts(WorkloadProfile profile, HoodieEngineContext context)
      * may result in OOM by making spark underestimate the actual input record sizes.
      */
     long averageRecordSize = averageBytesPerRecord(table.getMetaClient().getActiveTimeline()
-        .getTimelineOfActions(CollectionUtils.createSet(COMMIT_ACTION)).filterCompletedInstants(), config);
+        .getTimelineOfActions(CollectionUtils.createSet(COMMIT_ACTION, DELTA_COMMIT_ACTION)).filterCompletedInstants(), config);
```

Comment:
Could we be more strict here on the calculation of average bytes per record?  `deltacommit`s may contain log files and the number of bytes written for log files cannot be applied to writing parquet base files.  Let's only consider parquet file size here for record size estimation.

Reply:
There is a transformation ratio between log and parquet, but may not be very accurate.

Reply:
would you advise how can I distinguish/separate them?

Reply:
I think it's fine with your change now, just need to add some tests for the method that you might abstract out.

Reply:
as I commented previously, adding UT for func `assignInserts` is a little hard... 
any suggestion from your side?

Reply:
We can add a UT for `averageBytesPerRecord` I guess ?

Reply:
`averageBytesPerRecord`  is not changed and impacted as it just goes through the commits reversely to get the first eligible one to calculate the size.  this explains why #6864 did modify it either. 

here we go the existing  UTs for the function.  i know the value of UT while it is not easy to have one on the current code struct. 
https://github.com/apache/hudi/blob/master/hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/table/action/commit/TestUpsertPartitioner.java#L168-L190

Reply:
We can add a UT for `averageBytesPerRecord` to validate that both commit and delta_commit are included in the algorithm.